charts/kyverno/templates/networkpolicy.yaml

```diff
@@ -12,44 +12,89 @@ spec:
   - from:
-    {{- if or .Values.networkPolicy.namespaceExpressions .Values.networkPolicy.namespaceLabels }}
-      - namespaceSelector:
-        {{- with .Values.networkPolicy.namespaceExpressions }}
-          matchExpressions:
-            {{- toYaml . | nindent 10 }}
-        {{- end }}
-        {{- with .Values.networkPolicy.namespaceLabels }}
+    - podSelector:
           matchLabels:
-            {{- range $key, $value := . }}
-            {{ $key | quote }}: {{ $value | quote }}
-            {{- end }}
-        {{- end }}
+            app.kubernetes.io/name: prometheus
+  {{- if or .Values.networkPolicy.namespaceExpressions .Values.networkPolicy.namespaceLabels .Values.networkPolicy.podExpressions .Values.networkPolicy.podLabels }}
+    {{- if or .Values.networkPolicy.namespaceExpressions .Values.networkPolicy.namespaceLabels }}
+    - namespaceSelector:
+      {{- with .Values.networkPolicy.namespaceExpressions }}
+        matchExpressions:
+          {{- toYaml . | nindent 10 }}
+      {{- end }}
+      {{- with .Values.networkPolicy.namespaceLabels }}
+        matchLabels:
+          {{- range $key, $value := . }}
+          {{ $key | quote }}: {{ $value | quote }}
+          {{- end }}
+      {{- end }}
     {{- end }}
     {{- if or .Values.networkPolicy.podExpressions .Values.networkPolicy.podLabels }}
-        podSelector:
-          {{- with .Values.networkPolicy.podExpressions }}
-          matchExpressions:
-            {{- toYaml . | nindent 10 }}
-          {{- end }}
-          {{- with .Values.networkPolicy.podLabels }}
-          matchLabels:
-            {{- range $key, $value := . }}
-            {{ $key | quote }}: {{ $value | quote }}
-            {{- end }}
+    - podSelector:
+        {{- with .Values.networkPolicy.podExpressions }}
+        matchExpressions:
+          {{- toYaml . | nindent 10 }}
+        {{- end }}
+        {{- with .Values.networkPolicy.podLabels }}
+        matchLabels:
+          {{- range $key, $value := . }}
+          {{ $key | quote }}: {{ $value | quote }}
           {{- end }}
+        {{- end }}
     {{- end }}
     ports:
     - protocol: TCP
       port: 9443 # webhook access
+  {{- end }}
+
   # Allow prometheus scrapes for metrics
   {{- if .Values.metricsService.create }}
+    ports:
     - protocol: TCP
       port: {{ .Values.metricsService.port }}
+    {{- else }}
+    - protocol: TCP
+      port: 8000
   {{- end }}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-apiserver
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+  # Allow ingress from kube-apiserver on port 9443
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: kube-system
+        - podSelector:
+            matchLabels:
+              component: kube-apiserver
+              tier: control-plane
+      ports:
+        - port: 9443
+  # Allow egress to kube-apiserver on port 9443
```

Comment:
We should update this comment

Reply:
Does changing it to "If allowOCIAccess is true allow all egress traffic else allow egress to only kube-apiserver on port 9443" make sense?